cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -107,14 +108,29 @@ public void close() {
     }
   }
 
+  static String getBootstrapServers(Map<String, ?> configs) {
+    String listeners = String.valueOf(configs.get(KafkaConfig.ListenersProp()));
+    if (listeners != null && listeners.length() != 0) {
```

Comment:
This change(1) breaks backwards compatibility and (2) uses `null` as port if neither port nor listeners configs are specified.

### Breaks backwards compatibility:

In this new logic, if a user has both `listeners` and `port` configs specified, then the port config will be ignored now -- which was not the case in the existing logic.
Please ensure that 
1.  if `configs.get(KafkaConfig.PortProp())` is not `null`, use this as port (this is regardless of whether `listeners` has been specified in configs or not)
2. otherwise if `configs.get("listeners")` is not `null` and it has a non-empty `port`, use that as port
3. otherwise use the default hostname and port -- i.e. `localhost` and `9092`.

Please see https://github.com/linkedin/cruise-control/pull/1246#discussion_r449340518.

### May assign `null` as the port

If neither port nor listeners configs are specified, then `bootstrapServers` will be `localhost:null`.